charts/karmada/templates/pre-install-job.yaml

```diff
@@ -388,12 +388,12 @@ spec:
           mkdir -p /opt/configs
           mkdir -p /opt/certs
           cp -r -L /opt/mount/* /opt/configs/
-          openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout "/opt/certs/server-ca.key" -out "/opt/certs/server-ca.crt" -subj "/C=xx/ST=x/L=x/O=x/OU=x/CN=ca/emailAddress=x/"
-          openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout "/opt/certs/front-proxy-ca.key" -out "/opt/certs/front-proxy-ca.crt" -subj "/C=xx/ST=x/L=x/O=x/OU=x/CN=ca/emailAddress=x/"
+          openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:{{ printf `"%s"` .Values.certs.auto.rsaSize }} -keyout "/opt/certs/server-ca.key" -out "/opt/certs/server-ca.crt" -subj "/C=xx/ST=x/L=x/O=x/OU=x/CN=ca/emailAddress=x/"
```

Comment:
{{ .Values.certs.auto.rsaSize | quote }}

Reply:
hi, what is this ` | quote` intention?

Reply:
https://helm.sh/zh/docs/chart_template_guide/function_list/#quote-and-squote 

When the value read from .Values is turned into a string, it can be achieved by calling the quote template function.


to fix 

![image](https://github.com/karmada-io/karmada/assets/89568107/8a0935b8-3f4b-4509-9c8e-0920638d47c9)

Reply:
So it is more normalize to replace `printf  "%s" ...` with `... | quote` ?

There are still a few lines below that also use `printf  "%s" ...` functions, do we need to unify them all into `... | quote` ?

Reply:
sorry，my bad

{{ .Values.certs.auto.rsaSize }} should be used directly here. Using {{ .Values.certs.auto.rsaSize | quote }} will result in extra double quotes.

![image](https://github.com/karmada-io/karmada/assets/89568107/54dc1b6c-b9be-4c47-b224-f93ab00a6c52)


You can use helm template command to test locally

helm template ./charts/karmada/

![image](https://github.com/karmada-io/karmada/assets/89568107/673f9adc-9fa0-4272-a4b6-a0b06aae10eb)

Reply:
Now I totally get it. Thanks!